PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/MySQLLineMarkerProvider.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.intellij.runner;
+
+import com.intellij.codeInsight.daemon.GutterIconNavigationHandler;
+import com.intellij.codeInsight.daemon.LineMarkerInfo;
+import com.intellij.codeInsight.daemon.LineMarkerProvider;
+import com.intellij.lang.properties.psi.impl.PropertyImpl;
+import com.intellij.openapi.editor.markup.GutterIconRenderer;
+import com.intellij.openapi.module.Module;
+import com.intellij.openapi.module.ModuleUtil;
+import com.intellij.psi.PsiElement;
+import com.microsoft.azure.management.mysql.v2020_01_01.Server;
+import com.microsoft.azure.toolkit.intellij.link.base.LinkType;
+import com.microsoft.azure.toolkit.intellij.link.mysql.JdbcUrl;
+import com.microsoft.azure.toolkit.intellij.link.po.LinkPO;
+import com.microsoft.azure.toolkit.intellij.link.po.MySQLServicePO;
+import com.microsoft.azuretools.core.mvp.model.AzureMvpModel;
+import com.microsoft.azuretools.core.mvp.model.mysql.MySQLMvpModel;
+import com.microsoft.intellij.AzureLinkStorage;
+import com.microsoft.intellij.AzureMySQLStorage;
+import com.microsoft.intellij.helpers.AzureIconLoader;
+import com.microsoft.tooling.msservices.components.DefaultLoader;
+import com.microsoft.tooling.msservices.serviceexplorer.AzureIconSymbol;
+import com.microsoft.tooling.msservices.serviceexplorer.azure.mysql.MySQLNode;
+import org.apache.commons.lang3.StringUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.awt.event.MouseEvent;
+import java.util.Objects;
+import java.util.regex.Pattern;
+
+public class MySQLLineMarkerProvider implements LineMarkerProvider {
+
+    private static final Pattern MYSQL_PROPERTY_URL_VALUE = Pattern.compile("..");
+
+    @Override
+    public LineMarkerInfo getLineMarkerInfo(@NotNull PsiElement element) {
+        if (element instanceof PropertyImpl) {
+            PropertyImpl property = (PropertyImpl) element;
+            String value = property.getValue();
+            if (StringUtils.equals(property.getKey(), "spring.datasource.url") && StringUtils.startsWith(value, "${") && StringUtils.endsWith(value, "_URL}")) {
+                String envPrefix = value.substring(2, value.length() - 4);
```

Comment:
are you sure about value.length >?

Reply:
I suggest to move to a dedicated method to handle  the code:
```
StringUtils.equals(property.getKey(), "spring.datasource.url") && StringUtils.startsWith(value, "${") && StringUtils.endsWith(value, "_URL}")
```
and env prefix extraction

Reply:
yes. end with "URL}", so the value.length > 4